### PR TITLE
export-import improvements

### PR DESCRIPTION
**Pull request by mikehelmick**
## Proposed Changes

* De-duplication, don't schedule files more than once
* Fail files that age out (return 404 from download)
*

**Release Note**

```release-note
bug fixes: Export-importer job bug fixes. Prevents duplicate files from being scheduled. May require you to delete files from your `importfiles` table.
```

### CODE REVIEW

File: internal/exportimport/database/exportimport.go [hunk 1]
```diff
@@ -258,13 +261,20 @@ func (db *ExportImportDB) CreateFiles(ctx context.Context, ei *model.ExportImpor
...
 			result, err := tx.Exec(ctx, insertStmt, ei.ID, fname, now)
 			if err != nil {
 				return fmt.Errorf("error inserting filename: %v, %w", fname, err)
 			}
 			if result.RowsAffected() != 1 {
-				return fmt.Errorf("filename isnert failed: %v", fname)
+				logger.Warnw("attempted to insert duplicate file", "exportImportID", ei.ID)
+				continue
 			}
+			logger.Debugw("scheduled new epxort file for importing", "exportImportID", ei.ID, "filename", fname)
```
Comment (whaught) [inline internal/exportimport/database/exportimport.go:277]:
```suggestion
			logger.Debugw("scheduled new export file for importing", "exportImportID", ei.ID, "filename", fname)
```

File: internal/exportimport/import.go [hunk 2]
```diff
@@ -51,6 +56,17 @@ type SignatureAndKey struct {
 }
 
 func (s *Server) ImportExportFile(ctx context.Context, ir *ImportRequest) (*ImportResposne, error) {
+	// Special case - previous versions may have inserted the filename root as a file.
+	// If we find that, skip attempted processing and just mark as successful.
+	logging.FromContext(ctx).Debugw("does it float", "exportRoot", ir.exportImport.ExportRoot, "zipFilename", ir.file.ZipFilename)
```
Comment (whaught) [inline internal/exportimport/import.go]: Not sure if I understand `"does it float"`

File: internal/exportimport/scheduler.go [hunk 3]
```diff
@@ -82,8 +88,11 @@ func (s *Server) handleSchedule(ctx context.Context) http.HandlerFunc {
 			zipNames := strings.Split(string(bytes), "\n")
 			currentFiles := make([]string, 0, len(zipNames))
 			for _, zipFile := range zipNames {
+				if zipFile == "" {
```
Comment (whaught) [inline internal/exportimport/scheduler.go:91]:
```suggestion
				if len(strings.TrimSpace(zipFile)) == 0 {
```

File: internal/exportimport/import.go (hunk 2 above)
Comment (mikehelmick) [reply internal/exportimport/import.go]: debugging statement - removed in second commit.

**Review by whaught — APPROVED** [review]
/lgtm

File: internal/exportimport/scheduler.go [hunk 4]
```diff
@@ -82,8 +88,11 @@ func (s *Server) handleSchedule(ctx context.Context) http.HandlerFunc {
 			zipNames := strings.Split(string(bytes), "\n")
 			currentFiles := make([]string, 0, len(zipNames))
 			for _, zipFile := range zipNames {
+				if len(strings.TrimSpace(zipFile)) == 0 {
+					// drop blank lines.
+					continue
+				}
 				fullZipFile := fmt.Sprintf("%s%s", config.ExportRoot, strings.TrimSpace(zipFile))
```
Comment (jeremyfaller) [inline internal/exportimport/scheduler.go:95]: path.Join?

**Review by jeremyfaller — APPROVED** [review]

**Pull request merged**